### DISCUSSION TITLE
Data pipeline solution for jax?

### DISCUSSION THREAD

**Discussion by louis-shao** [jax-ml/jax#19436] [General]
Hi, if we use multi-GPU or multi-host training with jax, what's the expected data loading pipeline? When we use tensorflow we use the tf.data module with the tf distributed strategies, so when switched to jax we are using the same pipeline with dlpack to convert the result for jax to consume. Not sure if this is the optimal solution.

**Comment by wen020** [comment]
Do you currently know the best practices for implementing multi-host data parallelism with JAX?
